content/guides/10-minute-tutorial.md

```diff
@@ -792,12 +793,281 @@ Feature: Is it Friday yet?
+  def is_it_friday?(day)
+    answer = (day == 'Friday') ? 'TGIF' : 'Nope'
+  end
+end
+World FridayStepHelper
+
+Given("today is Sunday") do
+  @today = 'Sunday'
+end
+
+Given("today is Friday") do
+  @today = 'Friday'
+end
+
+# Then is a shared function
+When("I ask whether it's Friday yet") do
+  @actual_answer = is_it_friday?(@today)
+end
+
+# Then is a shared function
+Then("I should be told {string}") do |expected_answer|
+  expect(@actual_answer).to eq(expected_answer)
+end
+```
+{{% /block %}}
+
+Run Cucumber again:
+
+{{% block "javascript" %}}
+```shell
+......
+2 scenarios (2 passed)
+6 steps (6 passed)
+0m00.002s
+```
+{{% /block %}}
+
+{{% block "ruby" %}}
+```shell
+Feature: Is it Friday yet?
+  Everybody wants to know when it's Friday
+
+  Scenario: Sunday isn't Friday        # features/is_it_friday_yet.feature:4
+    Given today is Sunday              # features/step_definitions/stepdefs.rb:8
+    When I ask whether it's Friday yet # features/step_definitions/stepdefs.rb:17
+    Then I should be told "Nope"       # features/step_definitions/stepdefs.rb:22
+
+  Scenario: Friday is Friday           # features/is_it_friday_yet.feature:9
+    Given today is Friday              # features/step_definitions/stepdefs.rb:12
+    When I ask whether it's Friday yet # features/step_definitions/stepdefs.rb:17
+    Then I should be told "TGIF"       # features/step_definitions/stepdefs.rb:22
+
+2 scenarios (2 passed)
+6 steps (6 passed)
+0m0.040s
+```
+{{% /block %}}
+
+# Using variables and examples
+So, we all know that there are more days in the week than just Sunday and Friday. Let's update our scenario to use variables and evaluate more possibilities. We'll use variables and examples to evaluate Friday, Sunday, and anything else!
+
+Update the `is-it-friday-yet.feature` file. Notice how we go from `Scenario` to `Scenario Outline` when we start using `Examples`.
+```gherkin
+Feature: Is it Friday yet?
+  Everybody wants to know when it's Friday
+
+  Scenario Outline: Today is or is not Friday
+    Given today is <day>
+    When I ask whether it's Friday yet
+    Then I should be told <answer>
+
+  Examples:
+    | day | answer |
+    | "Friday" | "TGIF" |
+    | "Sunday" | "Nope" |
+    | "anything else!" | "Nope" |
+```
+
+{{% block "javascript" %}}
+Update the `stepdefs.js` file to use the `<day>` and `<answer>` variables.
+```javascript
+const assert = require('assert');
+const { Given, When, Then } = require('cucumber');
+
+function isItFriday(today) {
+  return (today === "Friday") ? "TGIF" : "Nope";
+}
+
+Given('today is {string}', function (givenDay) {
+  this.today = givenDay;
+});
+
+When('I ask whether it\'s Friday yet', function () {
+  this.actualAnswer = isItFriday(this.today);
+});
+
+Then('I should be told {string}', function (expectedAnswer) {
+  assert.equal(this.actualAnswer, expectedAnswer);
+});
+
+```
+{{% /block %}}
+
+{{% block "ruby" %}}
+Update the `stepdefs.rb` file to use the `<day>` and `<answer>` variables.
```

Comment:
Possibly one for another day, but ideally you'd have separate files for the helpers / step definitions (with the step defs file ending in `_steps.rb`

Reply:
I'm not sure what you mean here. Does Cucumber have opinionated docs on this somewhere that I could reference? I think that it would be good for the person using this tutorial to finish with a good understanding of what right looks like - assuming there is a formal opinion.

Reply:
I think from the hack-day we had. This is still a "new"-ish project. So I'd lean on the side of there isn't really a "right" way to do anything, as most of these docs are being created afresh.

Some of my comments have just come about from trying to adhere to the RSG but also trying to use conventions which are both common to ruby but also common to other OO languages (Not muddying files e.t.c.)

I only raised my points as a Comment not ReqChanges because I'm still reasonably raw in these areas, and other people may have stronger / more rounded opinions.

Anyways to come back to my original point. (It was sort of 2)

- Traditionally step definition files end in `_steps.rb`
- In most OO languages you retain one file for one "thing", i.e. a module / class / helper / controller...

For a 10min tutorial I guess it doesn't matter too much, but thought I'd throw it in that actually creating 2 small files still doesn't feel overkill for the 10min tutorial.

Reply:
@mikegfisher - No docs yet on using "helper" methods and classes, so it's fine if you keep them in one file for now (as I think that's what most people would to for a small project). As your project grows you might want to group steps into groups that are logical to your domain (and since this is a question that comes up regularly, I do want to clarify this in the docs). I'm ok with leaving it as is for now.

Reply:
Actually, I think the "step organization" page helps clarify this.